src/validate.js

```diff
@@ -2415,21 +2415,20 @@ function F_VALIDATE(disabledHL) {
 				&& isLimitOk(202)
 				&& address.isOkFor(202)) {
 				// Check other segments to be a drivable public segment
-				var foundPublicConnection = false;
-				for (var i = 0; i < nodeA.$otherSegmentsLen; i++) {
-					var otherSegment = nodeA.$otherSegments[i];
-					if (otherSegment.$rawSegment.isRoutable()) {
-						foundPublicConnection = true;
-						break;
-					}
-				}
-				if (!foundPublicConnection) {
-					// check node B
-					for (var i = 0; i < nodeB.$otherSegmentsLen; i++) {
-						var otherSegment = nodeB.$otherSegments[i];
-						if (otherSegment.$rawSegment.isRoutable()) {
+				// Second param is a segment to ignore as a valid public connection
+				var foundPublicConnection = checkPublicConnection(segment, null);
+				if(!foundPublicConnection){
+					// We might have a isolated segment. Could be a Restricted Gate
+					// See https://wazeopedia.waze.com/wiki/USA/Private_Installations#Specialty_Gate:_Restricted_Gate
+					if(nodeA.$otherSegmentsLen == 1 && nodeB.$otherSegmentsLen == 1){
+						// both sides are connected to just one private segment
+						var nodeASegment = nodeA.$otherSegments[0];
+						var nodeBSegment = nodeB.$otherSegments[0];
+						if (checkPublicConnection(nodeASegment, segment)
```

Comment:
I did not understand the use of this function. We checked both `nodeA` and `nodeB` connect just two segments (`nodeA.$otherSegmentsLen == 1 && nodeB.$otherSegmentsLen == 1`).

Would it be enough to just check that `nodeASegment.$type == RT_PRIVTE && nodeBSegment.$type == RT_PRIVATE` to exclude restricted gates?

Reply:
Example of a gate in the US:
![image](https://user-images.githubusercontent.com/63843/46284198-307bcd00-c577-11e8-8e75-559c0c1bc469.png)

I go and check if the private segments connected to this public segment are connected to a public segment on the other side(ignoring the segment being checked). 

There could be a case where the private segments are only connected to other private segments (excluding the possible isolated segment) and so the segment being checked IS a isolated public segment. 

Like a in this example I put together quick:
![image](https://user-images.githubusercontent.com/63843/46284911-d3cde180-c579-11e8-999e-59ff60bd2ac7.png)


If there are more than one private segments connected to this public segment on either side, it also isn't a gate (according to the US wiki) and so we can report it as a isolated segment.

Reply:
Sorry, I mean would it be enough to have just this condition:

    [...]
    if (!foundPublicConnection) {
        var nodeASegment = nodeA.$otherSegments[0];
        var nodeBSegment = nodeB.$otherSegments[0];
        // Exclude restricted gates
        if (nodeA.$otherSegmentsLen == 1 && nodeB.$otherSegmentsLen == 1
            && nodeASegment.$type == RT_PRIVTE && nodeBSegment.$type == RT_PRIVATE)
            foundPublicConnection = true;
    }
    [...]

Without calling `checkPublicConnection()` again? Sorry, I might be missing something...

Reply:
Ok, now I see your point. With those extra checks we will catch a lonely street segment among private roads, but not the restricted gates.

Well, I am ok with this change. But please note, that this check is already "slow" and adding extra loops makes it even slower. This all multiplied by the number of segments to validate, which might be up to few hundreds in the cities. It might affect the overall user experience, i.e. users might complain the Validator is slow overall.

Unfortunately, even with those extra loops, the check will not catch all the cases. For example, if we connect two street segment to each other and to private roads, the check will not report those two segments.

Does this extra check worth the slow down? I do not know. Potentially, there might be such lonely segments, but I am not sure how many of them out there on the map. If you think it worth it -- go ahead and merge the request. I will make a new release.

Reply:
Because of the speed issue I've tried to gate the check as much as I can (only check if the public segment is connected to a private road on both ends, if connected to multiple this won't trigger the extended check)

I  totally agree speed is an issue, I try to check London a few times (closest "big" city for me) to see if it's slow. For now, I think this is worth it and will merge it. But if people report issues with slowness this will one place to look into I agree!